### PR TITLE
Added handling no_proxy override through Proxies

### PR DESCRIPTION
**Pull request by gregory-shklover**

### CODE REVIEW

File: cpr/session.cpp [hunk 1]
```diff
@@ -180,6 +180,15 @@ void Session::prepareCommonShared() {
             curl_easy_setopt(curl_->handle, CURLOPT_PROXYPASSWORD, proxyAuth_.GetPassword(protocol));
         }
     }
+    // handle NO_PROXY override passed through Proxies object
+    // Example: Proxies{"no_proxy": ""} will override environment variable definition with an empty list
+    const std::string no_proxy[] = {"no_proxy", "NO_PROXY"};
```
Comment (COM8) [inline cpr/session.cpp:185]: Please use a `std::array` not a C-Style Array.

File: cpr/session.cpp [hunk 2]
```diff
@@ -180,6 +180,15 @@ void Session::prepareCommonShared() {
...
+    for (const auto& item : no_proxy) {
```
Comment (COM8) [inline cpr/session.cpp:186]: How about `std::any_of`?

File: test/proxy_tests.cpp [hunk 3]
```diff
@@ -86,6 +86,25 @@ TEST(ProxyTests, ReferenceProxySessionTest) {
     EXPECT_EQ(ErrorCode::OK, response.error.code);
 }
 
+if 0
+TEST(ProxyTests, NoProxyTest) {
```
Comment (COM8) [inline test/proxy_tests.cpp:90]: Please properly test your changes and let me know how I can verify them.

**Review by COM8 — CHANGES_REQUESTED** [review]
Thanks for the PR!
A few comments and how about an empty string for no proxy?

Please also add a docs PR here: https://github.com/libcpr/docs

File: cpr/session.cpp (hunk 2 above)
Comment (gregory-shklover) [reply cpr/session.cpp:186]: Does not seem that any_of would add readability.

File: test/proxy_tests.cpp (hunk 3 above)
Comment (gregory-shklover) [reply test/proxy_tests.cpp]: Added a test.
The test needs direct connection to the web + free proxy.

**Comment by gregory-shklover** [comment]
Added documentation pull request: https://github.com/libcpr/docs/pull/51/files

**Comment by COM8** [comment]
Thanks!

**Pull request merged**